src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/FairySouls.java

```diff
@@ -21,44 +21,273 @@
+	}
+
+	public void markAllAsMissing() {
+		if (!trackSouls) {
+			print(EnumChatFormatting.RED + "Fairy soul tracking is turned off, turn it on using /neu");
+			return;
+		}
+		if (allSoulsInCurrentLocation == null) {
+			print(EnumChatFormatting.RED + "No fairy souls found in your current world");
+			return;
+		}
+		foundSoulsInLocation.clear();
+		refreshMissingSoulInfo(true);
+
+		print(EnumChatFormatting.DARK_PURPLE + "Marked all fairy souls as not found");
+	}
+
+	private HashMap<String, Set<Integer>> getFoundSoulsForProfile() {
 		String profile = SBInfo.getInstance().currentProfile;
 		if (profile == null) {
-			if (loadedFoundSouls.containsKey(unknownProfile))
-				return loadedFoundSouls.get(unknownProfile);
+			if (allProfilesFoundSouls.containsKey(unknownProfile))
+				return allProfilesFoundSouls.get(unknownProfile);
 		} else {
 			profile = profile.toLowerCase(Locale.getDefault());
-			if (loadedFoundSouls.containsKey(unknownProfile)) {
-				HashMap<String, Set<Integer>> unknownProfileData = loadedFoundSouls.remove(unknownProfile);
-				loadedFoundSouls.put(profile, unknownProfileData);
+			if (allProfilesFoundSouls.containsKey(unknownProfile)) {
+				HashMap<String, Set<Integer>> unknownProfileData = allProfilesFoundSouls.remove(unknownProfile);
+				allProfilesFoundSouls.put(profile, unknownProfileData);
 				return unknownProfileData;
 			}
-			if (loadedFoundSouls.containsKey(profile)) {
-				return loadedFoundSouls.get(profile);
+			if (allProfilesFoundSouls.containsKey(profile)) {
+				return allProfilesFoundSouls.get(profile);
 			} else {
-				//create a new entry for this profile
+				// Create a new entry for this profile
 				HashMap<String, Set<Integer>> profileData = new HashMap<>();
-				loadedFoundSouls.put(profile, profileData);
+				allProfilesFoundSouls.put(profile, profileData);
 				return profileData;
 			}
 		}
 		return new HashMap<>();
 	}
 
-	public static void load(File file, Gson gson) {
-		loadedFoundSouls = new HashMap<>();
+	private static List<BlockPos> loadLocationFairySoulsFromConfig(String currentLocation) {
+		JsonObject fairySoulList = Constants.FAIRYSOULS;
+		if (fairySoulList == null) {
+			return null;
+		}
+
+		if (!fairySoulList.has(currentLocation) || !fairySoulList.get(currentLocation).isJsonArray()) {
+			return null;
+		}
+
+		JsonArray locations = fairySoulList.get(currentLocation).getAsJsonArray();
+		List<BlockPos> locationSouls = new ArrayList<>();
+		for (int i = 0; i < locations.size(); i++) {
+			try {
+				String coord = locations.get(i).getAsString();
+
+				String[] split = coord.split(",");
+				if (split.length == 3) {
+					String xS = split[0];
+					String yS = split[1];
+					String zS = split[2];
+
+					int x = Integer.parseInt(xS);
+					int y = Integer.parseInt(yS);
+					int z = Integer.parseInt(zS);
+
+					locationSouls.add(new BlockPos(x, y, z));
+				}
+			} catch (Exception ignored) {
```

Comment:
I think ignoring this exception is going to make debugging harder if a user ends up with a corrupted fairy soul list. printStackTrace should be fine.

Reply:
Agreed. This is part of the old code that I didn't spend the time to clean up. That said, I added the printStackTrace here and on line 330 as well so there's zero risk of regression.